src/DocBlock/Annotation.php

```diff
@@ -228,11 +228,17 @@ public function setTypes(array $types): void
      */
     public function getNormalizedTypes(): array
     {
-        $normalized = array_map(static fn (string $type): string => strtolower($type), $this->getTypes());
+        $typeExpression = $this->getTypeExpression();
+        if (null === $typeExpression) {
+            return [];
+        }
 
-        sort($normalized);
+        $normalizeTypeExpression = $typeExpression
+            ->mapTypes(static fn (TypeExpression $v) => new TypeExpression(strtolower($v->toString()), null, []))
+            ->sortTypes(static fn (TypeExpression $a, TypeExpression $b) => $a->toString() <=> $b->toString())
+        ;
 
-        return $normalized;
+        return $normalizeTypeExpression->getTypes();
```

Comment:
```suggestion
        return $normalizedTypeExpression->getTypes();
```

Reply:
same as in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8210#discussion_r1766242388